magma/backend/mlir/translation_unit.py

```diff
@@ -29,6 +29,8 @@ def _set_module_attrs(mlir_module: builtin.ModuleOp, opts: CompileToMlirOpts):
         lowering_options.append("disallowLocalVariables")
     if opts.omit_version_comment:
         lowering_options.append("omitVersionComment")
+    if opts.disallow_packed_struct_assignments:
```

Comment:
let's factor this out into a separate PR to add this option

Reply:
I'm not sure if this makes sense.  We need to compile with flatten_all_tuples=False to test it, otherwise the option isn't useful (since we wouldn't generate any packed struct assignments), so while we could add it as a separate PR, we wouldn't be able to have a meaningful test.  Adding it as a follow up PR also doesn't quite make sense either since this feature requires the option to be tested with certain tools.